packages/service-worker/src/low_level.ts

```diff
@@ -33,6 +33,16 @@ export interface UpdateActivatedEvent {
   current: {hash: string, appData?: Object};
 }
 
+/**
+ * An event emitted when a a resource can not be found anymore in the cache or on the server.
```

Comment:
I would rather make this more generic, because in the future there might be other scenarios that constitute an unrecoverable state. How about something like:

```suggestion
 * An event emitted when the version of the app used by the service worker to serve this client is
 * in a broken state that cannot be recovered from and a full page reload is required.
 *
 * For example, this can happen when a required resource cannot be found neither in the cache nor on
 * the server.
```